src/modules/brightness.rs

```diff
@@ -0,0 +1,343 @@
+            };
+
+        spawn(async move {
+            // make sure we have a value on startup and not have to wait for 1 interval
+            let _ = ctx.send(UiEvent::Refresh).await;
+
+            let mut partial_scroll: f64 = 0.0;
+
+            loop {
+                let event = tokio::select! {
+                    v = rx.recv() => v,
+                    _ = sleep(duration) => None,
+                };
+
+                let (mut percent, cur, max) =
+                    match Self::read_percentage(&client, &datasource, &default_resource_name).await
+                    {
+                        Ok(d) => d,
+                        Err(e) => {
+                            tracing::error!(?e, "Could not retrieve brightness levels");
+                            continue;
+                        }
+                    };
+
+                if let Some(UiEvent::AdjustBrightnessScroll(dy)) = event {
+                    partial_scroll += dy * scroll_speed;
+
+                    if partial_scroll.abs() >= 1.0 {
+                        let no_steps = partial_scroll.floor() as i32;
+                        partial_scroll -= partial_scroll.floor();
+
+                        let step_len = ((max as f64 / 100.0) as i32).max(1); // at least modify by 1 if max < 100
+
+                        let new_cur = cur - no_steps * step_len;
+
+                        let new_cur = new_cur.max(0).min(max); // not using .clamp to avoid panic in case max is ever < 0
+                        percent = new_cur as f64 / (max as f64) * 100.0;
+
+                        Self::write_brightness(
+                            &client,
+                            &datasource,
+                            &default_resource_name,
+                            new_cur,
+                        )
+                        .await;
+                    }
+                }
+
+                tx.send_update(BrightnessProperties {
+                    icon_name: icons.brightness_icon(percent).to_string(),
+                    screen_brightness: percent,
+                })
+                .await;
+            }
+        });
+
+        Ok(())
+    }
+
+    fn into_widget(
+        self,
+        context: WidgetContext<Self::SendMessage, Self::ReceiveMessage>,
+        _: &ModuleInfo,
+    ) -> color_eyre::Result<ModuleParts<Button>>
+    where
+        <Self as Module<Button>>::SendMessage: Clone,
+    {
+        let button_label = Label::builder()
+            .use_markup(true)
+            .justify(self.layout.justify.into())
+            .build();
+
+        let button = Button::new();
+        button.set_child(Some(&button_label));
+
+        let ctx = context.controller_tx.clone();
+
+        let scroll_controller = EventControllerScroll::new(EventControllerScrollFlags::VERTICAL);
+        scroll_controller.connect_scroll(move |_, _, dy| {
+            let ctx = ctx.clone();
+            spawn(async move { ctx.send(UiEvent::AdjustBrightnessScroll(dy)).await });
```

Comment:
There's an extension method to achieve this pattern:

```suggestion
            ctx.send_spawn(UiEvent::AdjustBrightnessScroll(dy));
```